Sources/Paywalls/Components/PaywallButtonComponent.swift

```diff
@@ -21,13 +21,34 @@ public extension PaywallComponent {
 
     struct ButtonComponent: PaywallComponentBase {
 
+        // swiftlint:disable nesting
+        public enum Action: Codable, Sendable, Hashable, Equatable {
+            case restorePurchases
+            case navigateTo(destination: Destination)
+            case navigateBack
+        }
+
+        public enum Destination: Codable, Sendable, Hashable, Equatable {
+            case customerCenter
+            case URL(url: URL, method: URLMethod)
+        }
+
+        public enum URLMethod: Codable, Sendable, Hashable, Equatable {
+            case inAppBrowser
+            case externalBrowser
+        }
+        // swiftlint:enable nesting
```

Comment:
@joshdholtz This is definitely not final, but wanted to gather some early feedback. Specifically: 
- Do we have a scheme for this already? (If so, where can I find it?)
- I see SwiftLint doesn't like the nesting, but it's helpful for namespacing. Otherwise all these enums would need substantially longer names. What is our usual approach?

Reply:
> Do we have a scheme for this already? (If so, where can I find it?)

We do not for this one! I had some Notion doc started where we were going to keep schemes and I'll link that to you in Slack

> I see SwiftLint doesn't like the nesting

It is a little agressive in its nesting requirements 😅  I usually put `// swiftlint:disable:next nesting` above each one so that I don't accidentally miss the "enable". Otherwise this looks fine!

Reply:
Thanks so much! Will continue on this path and let you know once it's ready for a real review 😄